wandb/sdk/wandb_verify.py

```diff
@@ -0,0 +1,493 @@
+    }
+    # create a file to save
+    filepath = "./test with_special-characters.txt"
+    f = open(filepath, "w")
+    f.write("test")
+    f.close()
+
+    with wandb.init(reinit=True, config=config) as run:
+        run_id = run.id
+        entity = run.entity
+        logged = True
+        try:
+            for i in range(1, 11):
+                run.log({"loss": 1.0 / i}, step=i)
+            log_dict = {"val1": 1.0, "val2": 2}
+            run.log({"dict": log_dict}, step=i + 1)
+        except Exception:
+            logged = False
+            failed_test_strings.append(
+                "Failed to log values to run. Contact W&B for support."
+            )
+
+        try:
+            run.log({"HT%3ML ": wandb.Html('<a href="https://mysite">Link</a>')})
+        except Exception:
+            failed_test_strings.append(
+                "Failed to log to media. Contact W&B for support."
+            )
+
+        wandb.save(filepath)
+
+    public_api = wandb.Api()
+    try:
+        prev_run = public_api.run("{}/{}/{}".format(entity, PROJECT_NAME, run_id))
+    except Exception:
+        failed_test_strings.append(
+            "Failed to access run through API. Contact W&B for support."
+        )
+        print_results(failed_test_strings, False)
+        return
+    for key, value in prev_run.config.items():
+        if config[key] != value:
+            failed_test_strings.append(
+                "Read config values don't match run config. Contact W&B for support."
+            )
+            break
+    if logged and (
+        prev_run.history_keys["keys"]["loss"]["previousValue"] != 0.1
+        or prev_run.history_keys["lastStep"] != 11
+        or prev_run.history_keys["keys"]["dict.val1"]["previousValue"] != 1.0
+        or prev_run.history_keys["keys"]["dict.val2"]["previousValue"] != 2
+    ):
+        failed_test_strings.append(
+            "History metrics don't match logged values. Check database encoding."
+        )
+
+    if logged and prev_run.summary["loss"] != 1.0 / 10:
+        failed_test_strings.append(
+            "Read summary values don't match expected value. Check database encoding, or contact W&B for support."
+        )
+    try:
+        read_file = prev_run.file(filepath).download(replace=True)
+    except Exception:
+        with wandb.init(
+            project=PROJECT_NAME, config={"test": "test direct saving"}
+        ) as run:
+            saved, status_code, _ = try_manual_save(api, filepath, run.id, run.entity)
+            if saved:
+                failed_test_strings.append(
+                    "Unable to download file. Check SQS configuration, topic configuration and bucket permissions."
+                )
+            else:
+                failed_test_strings.append(
+                    "Unable to save file with status code: {}. Check SQS configuration and bucket permissions.".format(
+                        status_code
+                    )
+                )
```

Comment:
I'm a bit confused why we're trying to save a file only if the download of the file fails...  Won't we have a save file error in the run above that calls `wandb.save`?  I guess we won't error to the user in that case because the save is happening in the backend?

Reply:
If understood right, wandb.save doesn't raise an exception if the save fails. So this is a work around. If the download fails, we try saving the file through URLs to see if the save worked at all.